R/groupingsets.R

```diff
@@ -105,6 +182,10 @@ groupingsets.data.table = function(x, j, by, sets, .SDcols, id = FALSE, jj, ...)
       missing.int64.by.cols = setdiff(int64.by.cols, by.set)
       if (length(missing.int64.by.cols)) r[, (missing.int64.by.cols) := bit64::as.integer64(NA)]
     }
+    if (!is.null(label))
+      for (total.var in intersect(setdiff(by, by.set), names(label.use))) {
+        r[, (total.var) := label.use[[total.var]]]
```

Comment:
We don't want to loop over `[.data.table`. I haven't looked closed but it feels it could be vectorized (handling of int64 2 lines above is indeed vectorized, therefore [ make sense there), but probably even simpler is to just loop with `set()`.

Reply:
Thanks, I've removed the loop.